src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -133,34 +133,137 @@ private void EmbedIfNeedTo(BoundExpression receiver, ImmutableArray<PropertySymb
             }
         }
 
+        public (
+            MethodSymbol method,
+            BoundExpression? receiverOpt,
+            ImmutableArray<BoundExpression> arguments,
+            ImmutableArray<int> argsToParamsOpt,
+            ImmutableArray<RefKind> argumentRefKindsOpt,
+            bool invokedAsExtensionMethod
+            ) InterceptCallAndAdjustArguments(BoundCall node)
+        {
+            var method = node.Method;
+            var receiverOpt = node.ReceiverOpt;
+            var arguments = node.Arguments;
+            var argsToParamsOpt = node.ArgsToParamsOpt;
+            var argumentRefKindsOpt = node.ArgumentRefKindsOpt;
+            var invokedAsExtensionMethod = node.InvokedAsExtensionMethod;
+
+            var interceptableLocation = node.InterceptableLocation;
+            if (this._compilation.GetInterceptor(interceptableLocation) is not var (interceptsLocationAttributeData, interceptor))
+            {
+                // The call was not intercepted.
+                return (method, receiverOpt, arguments, argsToParamsOpt, argumentRefKindsOpt, invokedAsExtensionMethod);
+            }
+
+            Debug.Assert(interceptableLocation != null);
+            if (!method.IsInterceptable)
+            {
+                // PROTOTYPE(ic): it was speculated that we could avoid work if we know the current method is not interceptable.
+                // i.e. use this as an early out before even calling Compilation.GetInterceptor.
+                // But by calling 'GetInterceptor' before this, we don't really avoid that work. Is that fine?
+                // PROTOTYPE(ic): eventually we probably want this to be an error but for now it's convenient to just warn
+                // so we can experiment with intercepting APIs that haven't yet been marked.
+                this._diagnostics.Add(ErrorCode.WRN_CallNotInterceptable, interceptsLocationAttributeData.AttributeLocation, method);
+            }
+
+            Debug.Assert(interceptor.Arity == 0);
+
+            // When the original call is to an instance method, and the interceptor is an extension method,
+            // we need to take special care to intercept with the extension method as though it is being called in reduced form.
+            var needToReduce = receiverOpt != null && interceptor.IsExtensionMethod;
+            var symbolForCompare = needToReduce ? ReducedExtensionMethodSymbol.Create(interceptor, receiverOpt!.Type, _compilation) : interceptor;
+            if (!MemberSignatureComparer.InterceptorsComparer.Equals(method, symbolForCompare))
```

Comment:
>if (!MemberSignatureComparer.InterceptorsComparer.Equals(method, symbolForCompare)) [](http://example.com/codeflow?start=12&length=83)

Do we need to check for `scoped` and `[UnscopedRef]` differences?

Reply:
Yeah, adding a note for it.